gateway/src/http_entrypoints.rs

```diff
@@ -304,23 +347,33 @@ impl GatewayApi for GatewayImpl {
         let apictx = rqctx.context();
         let PathSpComponent { sp, component } = path.into_inner();
         let sp_id = sp.into();
-        let sp = apictx.mgmt_switch.sp(sp_id)?;
-        let component = component_from_str(&component)?;
-        let slot = body.into_inner().slot;
-        let persist = query_params.into_inner().persist;
-
-        sp.set_component_active_slot(component, slot, persist).await.map_err(
-            |err| SpCommsError::SpCommunicationFailed { sp: sp_id, err },
-        )?;
-
-        Ok(HttpResponseUpdatedNoContent {})
+        let handler = async {
+            let sp = apictx.mgmt_switch.sp(sp_id)?;
+            let component = component_from_str(&component)?;
+            let slot = body.into_inner().slot;
+            let persist = query_params.into_inner().persist;
+
+            sp.set_component_active_slot(component, slot, persist)
+                .await
+                .map_err(|err| SpCommsError::SpCommunicationFailed {
+                    sp: sp_id,
+                    err,
+                })?;
+
+            Ok(HttpResponseUpdatedNoContent {})
+        };
+        apictx.latencies.instrument_dropshot_handler(&rqctx, handler).await
     }
 
     async fn sp_component_serial_console_attach(
         rqctx: RequestContext<Self::Context>,
         path: Path<PathSpComponent>,
         websocket: WebsocketUpgrade,
     ) -> WebsocketEndpointResult {
+        // TODO(eliza): I'm not sure whether there's a way to make
+        // `oximeter_instruments`'s HTTP latency tracker work with websockets
+        // requests? It would be nice to get the latency and any error returned
+        // prior to actually returning the websocket stream...
```

Comment:
It's also possible we could drop this endpoint. Today if anyone attaches to the serial console it happens via `faux-mgs`, not MGS proper, and it doesn't seem like something we need to expose to the control plane in general.

(I think this method's existence long predates any discussions of the tech port, support keys, etc. It's possible to use it via [`gateway-cli usart-attach ...`](https://github.com/oxidecomputer/omicron/blob/4a60d7843b0d5763021ce71b6a3410f8de0859e1/gateway-cli/src/main.rs#L194), but I strongly suspect that's never been used outside of its initial development.)

Reply:
Mmm, that makes sense --- perhaps we should open a new issue to discuss deprecating/removing it?

Reply:
Sure - filed https://github.com/oxidecomputer/omicron/issues/6437